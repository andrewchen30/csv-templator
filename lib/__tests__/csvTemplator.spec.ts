import CSVTemplator from '@/index';

describe('CSV Templator', () => {
  describe('parse schema ', () => {
    const successCases: {
      desc: string;
      template: string;
    }[] = [
      {
        desc: 'should parse data cell (Horizontal)',
        template: `
          | 'aaa' |
          | 'bbb' |
          | 'ccc' |
        `,
      },
      {
        desc: 'should parse data cell (Vertical)',
        template: `
          | 'aaa' | 'bbb' | 'ccc' |
        `,
      },
      {
        desc: 'should parse data cell (Mixed)',
        template: `
          | 'aaa' | 'bbb' |
          | 'ccc' | 'ddd' |
          | 'eee' | 'fff' |
        `,
      },
    ];

    it.each(successCases)('$desc', ({ template }) => {
      const templator = new CSVTemplator();
      templator.useTemplate(template);

      expect(templator._schema).toMatchSnapshot();
      expect(templator._logicColIndexes).toMatchSnapshot();
      expect(templator._logicRowIndexes).toMatchSnapshot();
    });
  });
});
