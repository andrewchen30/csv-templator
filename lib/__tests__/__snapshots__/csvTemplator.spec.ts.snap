// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV Templator parse schema  should parse data cell (Horizontal) 1`] = `
Array [
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 0,
      },
      "eval": "'aaa'",
      "renderPosition": Object {
        "col": 0,
        "row": 0,
      },
      "type": 1,
    },
  ],
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 1,
      },
      "eval": "'bbb'",
      "renderPosition": Object {
        "col": 0,
        "row": 1,
      },
      "type": 1,
    },
  ],
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 2,
      },
      "eval": "'ccc'",
      "renderPosition": Object {
        "col": 0,
        "row": 2,
      },
      "type": 1,
    },
  ],
]
`;

exports[`CSV Templator parse schema  should parse data cell (Horizontal) 2`] = `Set {}`;

exports[`CSV Templator parse schema  should parse data cell (Horizontal) 3`] = `Set {}`;

exports[`CSV Templator parse schema  should parse data cell (Mixed) 1`] = `
Array [
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 0,
      },
      "eval": "'aaa'",
      "renderPosition": Object {
        "col": 0,
        "row": 0,
      },
      "type": 1,
    },
    Object {
      "_positionInTemplate": Object {
        "col": 1,
        "row": 0,
      },
      "eval": "'bbb'",
      "renderPosition": Object {
        "col": 1,
        "row": 0,
      },
      "type": 1,
    },
  ],
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 1,
      },
      "eval": "'ccc'",
      "renderPosition": Object {
        "col": 0,
        "row": 1,
      },
      "type": 1,
    },
    Object {
      "_positionInTemplate": Object {
        "col": 1,
        "row": 1,
      },
      "eval": "'ddd'",
      "renderPosition": Object {
        "col": 1,
        "row": 1,
      },
      "type": 1,
    },
  ],
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 2,
      },
      "eval": "'eee'",
      "renderPosition": Object {
        "col": 0,
        "row": 2,
      },
      "type": 1,
    },
    Object {
      "_positionInTemplate": Object {
        "col": 1,
        "row": 2,
      },
      "eval": "'fff'",
      "renderPosition": Object {
        "col": 1,
        "row": 2,
      },
      "type": 1,
    },
  ],
]
`;

exports[`CSV Templator parse schema  should parse data cell (Mixed) 2`] = `Set {}`;

exports[`CSV Templator parse schema  should parse data cell (Mixed) 3`] = `Set {}`;

exports[`CSV Templator parse schema  should parse data cell (Vertical) 1`] = `
Array [
  Array [
    Object {
      "_positionInTemplate": Object {
        "col": 0,
        "row": 0,
      },
      "eval": "'aaa'",
      "renderPosition": Object {
        "col": 0,
        "row": 0,
      },
      "type": 1,
    },
    Object {
      "_positionInTemplate": Object {
        "col": 1,
        "row": 0,
      },
      "eval": "'bbb'",
      "renderPosition": Object {
        "col": 1,
        "row": 0,
      },
      "type": 1,
    },
    Object {
      "_positionInTemplate": Object {
        "col": 2,
        "row": 0,
      },
      "eval": "'ccc'",
      "renderPosition": Object {
        "col": 2,
        "row": 0,
      },
      "type": 1,
    },
  ],
]
`;

exports[`CSV Templator parse schema  should parse data cell (Vertical) 2`] = `Set {}`;

exports[`CSV Templator parse schema  should parse data cell (Vertical) 3`] = `Set {}`;
